tiledb/bioimg/converters/base.py

```diff
@@ -157,39 +130,62 @@ def to_tiledb(
         :param output_group_path: path to the TileDB group of arrays
         :param level_min: minimum level of the image to be converted. By default set to 0
             to convert all levels.
+        :param tiles: A mapping from dimension name (one of 'T', 'C', 'Z', 'Y', 'X') to
+            the (maximum) tile for this dimension.
+        :param preserve_axes: If true, preserve the axes order of the original image.
         """
         tiledb.group_create(output_group_path)
         with self._get_image_reader(input_path) as reader:
+            axes = reader.axes
             # Create a TileDB array for each level in range(level_min, reader.level_count)
             uris = []
             for level in range(level_min, reader.level_count):
-                uri = os.path.join(output_group_path, f"l_{level}.tdb")
+                # read metadata and image
+                metadata = reader.level_metadata(level)
                 image = reader.level_image(level)
-                canonical_image = reader.level_axes(level).transpose(image)
-                level_metadata = reader.level_metadata(level)
-                level_metadata["level"] = level
-                self._write_image(uri, canonical_image, level_metadata)
+                # determine axes and (optionally) transpose image to canonical axes
+                if preserve_axes:
+                    level_axes = axes
+                else:
+                    level_axes = axes.canonical(image)
+                    image = axes.transpose(image, level_axes)
+                # create TileDB array
```

Comment:
Why unwrapping the `_write_image`?

Reply:
Yeap I was thinking mostly about your second point. Define/Get the `schema` outside and pass it to `_write_image` were the logic of writing will be held.